smart-contracts/advanced/fevm-indexers.md

```diff
@@ -0,0 +1,92 @@
+3. The significant time required to retrieve data from the blockchain node.
+4. The complex interconnections between smart contracts are not easily identifiable.
+5. The substantial compute power needed to calculate advanced queries.
+
+Additionally, blockchain indexers provide a better developer experience by leveraging well-known API standards and query languages like GraphQL.
+
+## The Graph
+
+[The Graph](https://thegraph.com) is a decentralized protocol for indexing blockchain data. It enables developers to build and publish custom open APIs, known as subgraphs, that applications can query to retrieve blockchain data using GraphQL in a time-efficient manner.
+
+### Glossary
+
+- **Subgraphs**: Customizable schemas that define how to index data from specific blockchain smart contracts and events.
+- **GraphQL**: A query language that allows clients to request exactly the data they need, making data fetching more efficient.
+
+### Querying Subgraphs on Filecoin FEVM
+
+There are many ways to query existing subgraphs, including numerous well-known libraries for [JavaScript](https://thegraph.com/docs/en/querying/querying-from-an-application/) and [Python](https://thegraph.com/docs/en/querying/querying-with-python/). But even without any third-party tooling, querying a subgraph is no more complicated than querying [RPC nodes](https://docs.filecoin.io/reference/json-rpc). The only complexity is that you have to know the schema of the subgraph beforehand, similar to knowing SQL database tables and columns before being able to query them. Luckily, The Graph provides several ways to discover the subgraph schema. The most convenient one is called the ["Playground"](https://graphql.org/blog/2020-04-03-graphiql-graphql-playground/), and it is available upon a GET request to the subgraph query URL. Alternatively, you may use the discovery method that exists on every subgraph, called the [Introspection Query](https://graphql.org/learn/introspection/).
+
+### Developing Subgraphs on Filecoin FEVM
+
+Developing a subgraph requires specialized knowledge that can be obtained through [The Graph Academy](https://thegraph.academy). Although it is not very complicated, the exact structure of a subgraph is an ever-changing process, so the details will stay out of scope for this documentation.
+
+### Deploying Subgraphs
+
+Just as with database data queried through SQL, subgraphs have to be stored somewhere. You may run a self-hosted instance as described in [The Graph Academy examples](https://thegraph.academy/developers/local-development/) and deploy a subgraph there. However, as with RPC nodes and databases, running subgraphs locally in production is not recommended from an uptime standpoint. For hosting the subgraph, it is reasonable to use online web services such as AWS or refer to professional subgraph providers such as [Protofire (aka Glif Nodes)](https://api.node.glif.io).
+
+#### Example: Deploying a Subgraph with Glif Nodes (Protofire)
```

Comment:
I'll let others decide as I don't know the convention here, but this largely feels like docs that Glif should own and should be on their site.  I would imagine filecoin docs just surfaces that this is available but then links off to their docs with the tutorial/example steps.

Basically, I assume we aren't going to do this for each subgraph provider.  We'd instead just have a list of the providers and where you can go to learn more.

Reply:
Well, I understand your concern, it looks pretty valid. 
We are currently working on our version of the docs and it might make sense to migrate off eventually, although since there are not that many other teams competing over providing this service maybe it might make sense to keep it here for now, which would allow for two things:
1) Onboard more developers who might want to jump into Filecoin subgraphs before devcon
2) Grant some time for us to develop our own docs and eventually migrate it there, because right now there is no place on our end to store this, and despite the fact we have a solid UI, the UX kinda... in development.
But in like month or so - I'm totally up to migrating this. Or if there are new public subgraph providers emerge and want to have a bit more equal competition, like we have for RPCs.

Anyways, will let the repo maintainers decide.

Reply:
Makes sense.  Inlining this for now seems great to me until there is a better home or other competing providers.  Good stuff - way to be pragmatic!